src/backends/c_emitter.h

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+
+Author: Jared Roesch
+*/
+#pragma once
+
+#include <iostream>
+#include "kernel/environment.h"
+#include "backend.h"
+#include "simple_expr.h"
+
+namespace lean  {
+    // static const char * LEAN_OBJ_TYPE = "lean::obj";
+    // static const char * LEAN_ERR = "lean::runtime_error";
+
+    class c_emitter {
```

Comment:
This class is currently empty, but the goal is to clean up the backend by moving most of the formatting code to the emitter, and then just having the backend use the emitter to write out the actual strings.